### PR TITLE
[Refactor] Move the Relay parts of an infinite scroll view into it's own files - #trivial

### PR DESCRIPTION
**Pull request by orta**
As I need to add another infinite scroll section. This separates out the responsibility of making the grid, and of using it with Relay.

---

As I hit https://github.com/artsy/emission/issues/505 I fixed it too, and added tests for all the missing files.

### CODE REVIEW

File: dangerfile.ts [hunk 1]
```diff
@@ -73,14 +73,17 @@ const correspondingTestsForAppFiles = touchedAppOnlyFiles.map(f => {
...
 if (testFilesThatDontExist.length > 0) {
   const callout = acceptedNoTests ? warn : fail
   const output = `Missing Test Files:
-    ${testFilesThatDontExist.map(f => `  - [] \`${f}\``).join("\n")}
 
-    If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
-  `
+${testFilesThatDontExist.map(f => `- \`${f}\``).join("\n")}
+
+If these files are supposed to not exist, please update your PR body to include "Skip New Tests".`
```
Comment (orta) [inline dangerfile.ts]: this just updates the markdown produced, it was pretty ugly when printed

File: dangerfile.ts [hunk 2]
```diff
@@ -73,14 +73,17 @@ const correspondingTestsForAppFiles = touchedAppOnlyFiles.map(f => {
 
 // New app files should get new test files
 // Allow warning instead of failing if you say "Skip New Tests" inside the body, make it explicit.
-const testFilesThatDontExist = correspondingTestsForAppFiles.filter(f => fs.existsSync(f))
+const testFilesThatDontExist = correspondingTestsForAppFiles
+                                 .filter(f => !f.includes("index-tests.tsx")) // skip indexes
+                                 .filter(f => !fs.existsSync(f))
```
Comment (orta) [inline dangerfile.ts]: The check for whether the file exists was wrong it was filtering only when it did exist

File: package.json [hunk 3]
```diff
@@ -33,7 +33,7 @@
       "^.+\\.js$": "<rootDir>/node_modules/babel-jest",
       ".(ts|tsx)": "<rootDir>/node_modules/ts-jest/preprocessor.js"
     },
-    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(ts|tsx|js)$",
+    "testRegex": "/__tests__/.*-tests.(ts|tsx|js)$",
```
Comment (orta) [inline package.json]: I added some fixtured data in the tests, which caused it to be recognised by jest

Comment (mzikherman) [inline package.json]: 👍

**Pull request merged**